web/models.py

```diff
@@ -299,6 +310,14 @@ def generate_activations(
     def is_healthy(self) -> bool:
         return self._state.is_healthy()
 
+    def is_timed_out(self, timeout):
+        return self._state.is_timed_out(timeout)
+    
+    def last_generation(self):
+        last_generation_query = db.select(ModelInstanceGeneration).where(ModelInstanceGeneration.model_instance_id == self.id).order_by(ModelInstanceGeneration.created_at.desc())
```

Comment:
If the above comment holds, this should also change to the following right?
```suggestion
        last_generation_query = db.select(ModelInstanceGeneration).where(ModelInstanceGeneration.model_instance_id == self.id).order_by(ModelInstanceGeneration.updated_at.desc())
```